pkg/services/dashboards/database/database_public_dashboard.go

```diff
@@ -0,0 +1,126 @@
+package database
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/util"
+	"github.com/grafana/grafana/pkg/util/errutil"
+)
+
+func generateNewPublicDashboardUid(sess *sqlstore.DBSession) (string, error) {
+	for i := 0; i < 3; i++ {
+		uid := util.GenerateShortUID()
+
+		exists, err := sess.Where("uid=?", uid).Get(&models.PublicDashboard{})
+		if err != nil {
+			return "", err
+		}
+
+		if !exists {
+			return uid, nil
+		}
+	}
+
+	return "", models.ErrPublicDashboardFailedGenerateUniqueUid
+}
+
+// retrieves public dashboard configuration
+func (d *DashboardStore) GetPublicDashboardConfig(orgId int64, dashboardUid string) (*models.PublicDashboardConfig, error) {
+	// get global dashboard config
+	var dashRes []*models.Dashboard
+
+	if dashboardUid == "" {
+		return nil, models.ErrDashboardIdentifierNotSet
+	}
+
+	err := d.sqlStore.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		return sess.Where("org_id = ? AND uid= ?", orgId, dashboardUid).Find(&dashRes)
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(dashRes) == 0 {
+		return nil, models.ErrDashboardNotFound
+	}
+
+	pdc := &models.PublicDashboardConfig{
+		IsPublic: dashRes[0].IsPublic,
+	}
+
+	// get public dashboards
+	var pdRes []*models.PublicDashboard
+	err = d.sqlStore.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
```

Comment:
Is this querying the dashboards table instead of the dashboard_public_config table?

Reply:
No, it's querying from dashboard_public_config and finds the table name from models.PublicDashboard.TableName()